be/src/runtime/data_stream_sender.cpp

```diff
@@ -706,23 +706,33 @@ Status DataStreamSender::serialize_chunk(const vectorized::Chunk* src, ChunkPB*
     if (_compress_codec != nullptr && uncompressed_size > 0) {
         SCOPED_TIMER(_compress_timer);
 
-        // Try compressing data to _compression_scratch, swap if compressed data is smaller
-        int max_compressed_size = _compress_codec->max_compressed_len(uncompressed_size);
+        if (_compress_codec->type() == CompressionTypePB::LZ4_FRAME ||
+            _compress_codec->type() == CompressionTypePB::ZSTD || _compress_codec->type() == CompressionTypePB::LZ4) {
+            Slice compressed_slice;
+            Slice input(dst->data());
+            _compress_codec->compress(input, &compressed_slice, true, uncompressed_size, nullptr,
```

Comment:
the same code appears in tablet_sink.cpp, suggest to put into the same function, then invoke this function.